actor-server/actor-core/src/main/scala/im/actor/api/rpc.scala

```diff
@@ -73,21 +79,24 @@ package object rpc extends {
     authorizedAction.getOrElse(DBIO.successful(Result.UserNotAuthorized))
   }
 
+  //we may remove recover call here.
   def toResult[R](authorizedFuture: MaybeAuthorized[Future[RpcError \/ R]])(implicit ec: ExecutionContext): Future[RpcError \/ R] =
     recover(authorizedFuture.getOrElse(Future.successful(Result.UserNotAuthorized)))
 
```

Comment:
currently `recoverFailure` in Base.scala looks following way:

```
final def recoverFailure[A <: RpcResponse]: scala.PartialFunction[Throwable, HandlerResult[A]] = 
  onFailure.andThen({ e => -\/(e) })
```

we can remove `recover` from here if we declare `recoverFailure` following way:
```
final def recoverFailure[A <: RpcResponse]: scala.PartialFunction[Throwable, HandlerResult[A]] =
  (im.actor.api.rpc.recoverCommon orElse onFailure) andThen({ e => -\/(e) })
```
But the bad part is - we reference server code from api here.